### PR TITLE
Remove deprecated auto-build CLI option

### LINKED ISSUE #23360: Remove deprecated `auto-build` option
### Description

I believe it's time to remove the `auto-build` CLI flag from Quarkus distro as the functionality was completely superseded by providing the `--optimized` flag. The option has been deprecated since Keycloak 19.

### PR DESCRIPTION
**Pull request by mabartos**
Closes #23360

@vmuzikar I believe it's time to get rid of it and clean up the config a little bit, right? :))

### CODE REVIEW

File: docs/documentation/upgrading/topics/keycloak/changes-23_0_0.adoc [hunk 1]
```diff
@@ -72,3 +72,12 @@ New code example:
     </button>
 </div>
 ----
+
+= The deprecated `auto-build` CLI option was removed
+
+The `auto-build` CLI option has been marked as deprecated for a long time.
+In this release, it was completely removed, and it is no longer supported.
+
+If you are using this option, you can leverage the `--optimized` CLI option, which provides a similar functionality.
```
[the remark this answers is not in the document]
Comment (mabartos) [reply docs/documentation/upgrading/topics/keycloak/changes-23_0_0.adoc:81]: Depends on how you think about it as it's kinda connected in the end. But you're right, doesn't have to be mentioned here. Thanks

**Review by pedroigor — DISMISSED** [review]

File: quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/StartAutoBuildDistTest.java
Comment (vmuzikar) [inline quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/StartAutoBuildDistTest.java:1]: I don't think we should remove this test class completely. I believe arts of it are still valid as they focus on `--optimized` flag and build process in general etc.

**Review by vmuzikar — CHANGES_REQUESTED** [review]

File: quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/StartAutoBuildDistTest.java
Comment (pedroigor) [reply quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/StartAutoBuildDistTest.java:1]: Ohh, that is true.

**Review by pedroigor — APPROVED** [review]

**Comment by mabartos** [comment]
@vmuzikar Can you check it, please?

**Review by vmuzikar — APPROVED** [review]
LGTM, thanks

**Pull request merged**